### PR TITLE
CB-19604 E2E Marketplace image tests for datalake and datahub

### PR DESCRIPTION
**Pull request by daszabo**
See detailed description in the commit message.

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/sdx/SdxMarketplaceImageTests.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        AtomicReference<String> selectedImageID = new AtomicReference<>();
+
+        SdxDatabaseRequest sdxDatabaseRequest = new SdxDatabaseRequest();
+        sdxDatabaseRequest.setAvailabilityType(SdxDatabaseAvailabilityType.NONE);
+
+        String imgCatalogKey = "mp-img-cat";
+        testContext
+                .given(imgCatalogKey, ImageCatalogTestDto.class)
+                .withName(imgCatalogKey)
+                .withUrl("https://gist.githubusercontent.com/daszabo/6b53c1a413d9b6cb847b9ad786e7bbf5/raw/0edb66302b7f485527b3bc03fdff1d68695d354c/" +
```
Comment (aszegedi) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/sdx/SdxMarketplaceImageTests.java:71]: Could you replace this GIST reference with a versioned one in this project at https://github.com/hortonworks/cloudbreak/tree/master/integration-test/src/main/resources. Please create a new sub folder then move the `v3-marketplace-image-catalog.json` to here, like we have for blueprints.

**Review by aszegedi — CHANGES_REQUESTED** [review]

**Comment by aszegedi** [comment]
Please add commit message for your commit with a brief description about the goal and reason of this change. This should be appeared as the PR description too.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/marketplace/AzureImageTermsSignerService.java [hunk 2]
```diff
@@ -69,18 +72,19 @@ public boolean isSigned(String subscriptionId, AzureMarketplaceImage azureMarket
...
 
         try {
-            AzureImageTerms azureImageTerms = azureRestOperationsService.httpGet(signUri, AzureImageTerms.class, token);
+            AzureImageTerms azureImageTerms = azureRestOperationsService.httpGet(agreementUri, AzureImageTerms.class, token);
             LOGGER.debug("Image terms and conditions received for image {} is : {}", azureMarketplaceImage, azureImageTerms);
 
             azureImageTerms.getProperties().setAccepted(true);
 
-            AzureImageTerms responseImageTerms = azureRestOperationsService.httpPut(signUri, azureImageTerms, AzureImageTerms.class, token);
+            AzureImageTerms responseImageTerms = azureRestOperationsService.httpPost(signUri, null, AzureImageTerms.class, token);
```
Comment (gergopapi2) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/marketplace/AzureImageTermsSignerService.java:87]: I tried locally and for me the POST endpoint with the above URL gives an error:
`Bad Request({"error":{"code":"BadRequest","message":"You must accept the legal terms through the Azure portal (http://go.microsoft.com/fwlink/?LinkId=534873). Correlation Id: '1bdf2fe2-542a-47ce-9a14-80abe382499d'."}})`

I also remember that I had to do it over the PUT endpoint. I quickly tested it via the command `az rest` and agreement accept and decline did work.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXMarketplaceImageTests.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        String imageSettings = resourcePropertyProvider().getName();
+        String distrox = resourcePropertyProvider().getName();
+        AtomicReference<String> selectedImageID = new AtomicReference<>();
+        CloudProvider cloudProvider = testContext.getCloudProvider();
+
+        String imgCatalogKey = "mp-img-cat";
+        testContext
+                .given(imgCatalogKey, ImageCatalogTestDto.class)
+                .withName(imgCatalogKey)
+                .withUrl("https://gist.githubusercontent.com/daszabo/6b53c1a413d9b6cb847b9ad786e7bbf5/raw/0edb66302b7f485527b3bc03fdff1d68695d354c/" +
```
Comment (gergopapi2) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXMarketplaceImageTests.java:60]: Would be better to have an official marketplace image catalog, up in s3, next to the test image catalog.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/sdx/SdxMarketplaceImageTests.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    private ImageCatalogTestClient imageCatalogTestClient;
+
+    @Test(dataProvider = TEST_CONTEXT)
+    @UseSpotInstances
+    @Description(
+            given = "there is a running Cloudbreak",
+            when = "a valid SDX create request is sent (latest Marketplace Image)",
+            then = "SDX should be available AND deletable"
+    )
+    public void testSDXWithMarketplaceImageCanBeCreatedSuccessfully(TestContext testContext) {
```
Comment (gergopapi2) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/sdx/SdxMarketplaceImageTests.java:52]: Would this mean a brand new test would be introduced for marketplace testing? 
Just from pure test cost savings, would it be possible to alter one of our existing tests, to use marketplace? Or, change one of the azure suits (api or longrunning) to use marketplace images.

**Review by gergopapi2 — CHANGES_REQUESTED** [review]
Let's find out locally, how come the POST method works for you and does not for me, and vica versa the PUT works for me and not for you. Maybe there is a difference in how marketplace images are published now? Let's put this PR on hold until then.

**Review by aszegedi — APPROVED** [review]

**Comment by daszabo** [comment]
> Let's find out locally, how come the POST method works for you and does not for me, and vica versa the PUT works for me and not for you. Maybe there is a difference in how marketplace images are published now? Let's put this PR on hold until then.

Fixed the signing request based on your draft PR

**Pull request merged**